nwb_conversion_tools/baserecordingextractorinterface.py

```diff
@@ -75,14 +74,16 @@ def convert_data(self, nwbfile, metadata_dict: None, stub_test=False):
         else:
             stub_recording_extractor = self.recording_extractor
 
-        if metadata_dict is not None and 'Ecephys' in metadata_dict and 'subset_channels' in metadata_dict['Ecephys']:
-            recording_extractor = se.SubRecordingExtractor(stub_recording_extractor,
-                                                           channel_ids=metadata_dict['Ecephys']['subset_channels'])
+        if metadata is not None and 'Ecephys' in metadata and 'subset_channels' in metadata['Ecephys']:
+            recording_extractor = se.SubRecordingExtractor(
+                stub_recording_extractor,
+                channel_ids=metadata['Ecephys']['subset_channels']
```

Comment:
I don't really like `subset_channels` being in the metadata dictionary. Can this be set as an attribute in the `__init__`? (set to `None` by default). This would allow it to be easily changed programmatically if the user needs to.